In [9]:
from linear_algebra import *
from old_linear_algebra import RandomMatrix as OldRandomMatrix

## Time comparison

In [5]:
%timeit OldRandomMatrix(order=1000) # Calculating the time to be reqd to execute this cell

649 ms ± 8.85 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


Hence ```OldRandomMatrix``` takes on an average $649$ mili seconds to generate one $1000\times 1000$ matrix!

In [11]:
%timeit RandomMatrix(order=1000)

19 ms ± 1.68 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)


In [8]:
%timeit HaarDistributedUnitary(size=1000)

649 ms ± 172 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
